synapse/push/httppusher.py

```diff
@@ -342,9 +346,83 @@ async def _process_one(self, push_action: HttpPushAction) -> bool:
                     await self._pusherpool.remove_pusher(self.app_id, pk, self.user_id)
         return True
 
-    async def _build_notification_dict(
-        self, event: EventBase, tweaks: Dict[str, bool], badge: int
-    ) -> Dict[str, Any]:
+    async def dispatch_push(
+        self,
+        content: JsonDict,
+        tweaks: Optional[Mapping[str, SimpleJsonValue]] = None,
+        default_payload: Optional[JsonMapping] = None,
+    ) -> Union[bool, Iterable[str]]:
+        """Send a notification to the registered push gateway, with `content` being
+        the content of the `notification` top property specified in the spec.
+        If specified, the `devices` property will be overridden with device-specific
+        information for this pusher.
```

Comment:
I'm tripping over the "If specified" which makes it sound like it is sometimes valid to provide this and sometimes not

```suggestion
        Note that the `devices` property will be added with device-specific
        information for this pusher.
```